#Pricing Strategy & Psychological Pricing Analysis
**Course:** MGMT 52610 — Data & AI-Driven Marketing  
**Dataset:** Tropicana Orange Juice Sales Panel  
**Topics:** Log-log regression, price elasticity, optimal pricing, psychological pricing patterns

In [ ]:
# Import required libraries
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import statsmodels.formula.api as smf

In [ ]:
# Load data
df = pd.read_csv("data/oj_tropicana.csv")

Part I:

1. Estimate a log-log sales regression model for the Tropicana brand to analyze the effect of advertising (measured by feat in the data) on sales and price elasticity

In [ ]:
reg = smf.ols("np.log(sales) ~ np.log(price) * feat", data=df).fit()
print(reg.summary().tables[1])

                         coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------------
Intercept             11.1145      0.039    284.956      0.000      11.038      11.191
np.log(price)         -2.0384      0.036    -56.802      0.000      -2.109      -1.968
feat                   1.8797      0.079     23.864      0.000       1.725       2.034
np.log(price):feat    -1.4567      0.086    -16.921      0.000      -1.625      -1.288


2. Report the effect of feat on sales. Hint: Think about how to interpret the corresponding coefficient in the regression when describing the effect.

The coefficient on feat is 1.8797
This means that at a fixed price, advertising raises log(sales) by 1.8797 compared to weeks with no advertising.
Converting this into a percentage effect using formula:
"% change in sales" = (e^(1.8797) - 1) x 100

In [ ]:
coef_feat = 1.8797
percent_increase = (np.exp(coef_feat) - 1) * 100
percent_increase

np.float64(555.153910561072)


So, when Tropicana is advertised, sales are roughly 555% higher compared to weeks when it is not advertised. In other words, advertising has a substantial positive effect on Tropicana sales.

3. Calculate and report the price elasticity for when the brand it is not featured vs. when it is featured. Discuss how advertising affects sales and price elasticity.

When Tropicana is not featured, the price elasticity of demand (PED) is given by the coefficient on log(price): −2.04.
When Tropicana is featured, the PED is the sum of the coefficient on log(price) and the interaction term log(price)×feat: −2.04 + (−1.46) = −3.50.
So, when featured, Tropicana’s demand becomes more elastic - it's elasticity drops by about 1.5, moving from −2.04 to −3.50. Advertising both raises sales levels and increases price sensitivity. We can conclude that when Tropicana is advertised, customers buy more overall, but their quantity demanded responds more strongly to price changes than it would without advertising.

4. Use the optimal pricing formula (on slide 6 of lecture 6) to calculate the optimal retail price (per Carton) for Tropicana when the it is not advertised vs when it is advertised. Discuss how advertising impacts the brand’s pricing strategy.

Profit-maximizing price for OJ given PED of $\gamma$:

price = ($\gamma$ / (1 + $\gamma$)) * cost

From the above calculations, we found that:

$\gamma$ when not advertised = -2.0

$\gamma$ when advertised = -3.5


Solving for optimal retail price when not advertised:
price = (-2.0 / (1 - 2.0)) * cost
      = (-2 / -1) * cost
      = 2cost
Solving for optimal retail price when advertised:
price = -3.5 / (1 - 3.5)) * cost
      = (-3.5 / -2.5) * cost
      = 1.4cost
Example
If cost = $1,
optimal price when not advertised = $2.00
optimal price when advertised = $1.40
When Tropicana is not advertised, demand is less elastic (−2.0), and Tropicana would charge a higher markup of about about 2× cost to maximize profit. So if, for example, cost is 1.00 USD, optimal price would be 2.00 USD
When Tropicana is advertised, demand becomes more elastic (−3.50), and Tropicana would charge a lower markup of about 1.40× cost to maximize profit. So if, for example, cost is 1.00 USD, optimal price would be 1.40 USD
Therefore, advertising increases sales but also increases consumers’ price sensitivity, so the profit-maximizing price is lower in advertised weeks. The brand should advertise and reduce price simultaneously to maximize profit when advertising.

Part II

5. Create three new variables variables (price_int, price_decimal1, price_decimal2) to represent the three digits of price in unit of dollars. For example, for price = 2.87, price_int = 2, price_decimal1 = 0.8, and price_decimal2 = 0.07.

In [ ]:
df["price_int"] = np.floor(df["price"])
df["price_decimal1"] = (np.floor(df["price"]*10) % 10) / 10
df["price_decimal2"] = (np.floor(df["price"]*100) % 10) / 100

# Look at the some sample rows
print(df[["price", "price_int", "price_decimal1", "price_decimal2"]].sample(10))

price  price_int  price_decimal1  price_decimal2
4077   3.19        3.0             0.1            0.09
8963   1.99        1.0             0.9            0.09
7593   3.19        3.0             0.1            0.09
9085   2.75        2.0             0.7            0.05
1103   3.59        3.0             0.5            0.09
7408   2.94        2.0             0.9            0.04
5567   3.19        3.0             0.1            0.09
8947   3.07        3.0             0.0            0.07
34     3.59        3.0             0.5            0.09
8342   2.67        2.0             0.6            0.07


6. Tabulate the frequency of the unique values of the three new price variables. Plot histograms of the three new variables. Comment on the patterns you discover in the three distributions (focus on the two decimal price variables).

In [ ]:
# Frequency tables
print(df["price_int"].value_counts().sort_index())
print(df["price_decimal1"].value_counts().sort_index())
print(df["price_decimal2"].value_counts().sort_index())

price_int
1.0    1353
2.0    4533
3.0    3763
Name: count, dtype: int64
price_decimal1
0.0     262
0.1    1307
0.2     540
0.3    1121
0.4     473
0.5    1173
0.6     693
0.7     758
0.8     559
0.9    2763
Name: count, dtype: int64
price_decimal2
0.00     207
0.01     167
0.02      63
0.03     333
0.04     414
0.05     105
0.06     595
0.07     500
0.08      97
0.09    7168
Name: count, dtype: int64


In [ ]:
import seaborn as sns
import matplotlib.pyplot as plt

plt.figure(figsize=(12,4))

# price_int
plt.subplot(1,3,1)
sns.histplot(df["price_int"], bins=3)
plt.xticks([1,2,3])
plt.title("price_int")

# price_decimal1
plt.subplot(1,3,2)
sns.histplot(df["price_decimal1"], bins=10)
plt.title("price_decimal1")

# price_decimal2
plt.subplot(1,3,3)
sns.histplot(df["price_decimal2"], bins=10)
plt.title("price_decimal2")

plt.tight_layout()
plt.show()

price_int: Prices cluster at $1, $2, and $3, which is typical for orange juice
price_decimal1 (first decimal digit): The distribution is not uniform. Values around 0.8–0.9 appear far more often, showing that stores frequently price items at .79, .89, .99, etc., rather than round decimals like .10 or .20.
price_decimal2 (second decimal digit): The distribution is extremely skewed. 0.09 dominates, indicating that most prices end in 9 cents. Other digits occur far less often.

Interpretation: The results suggest that while the dollar amounts vary normally, the decimal digits do not. The first decimal digit concentrates near 0.8 and 0.9, and the second digit is dominated by 0.09. This pattern indicates that most prices end in .89 or .99, rather than being spread evenly across decimal values. This reflects a common retail pricing strategy where prices are intentionally set just below the next dollar amount because shoppers tend to perceive these prices as meaningfully lower.

7. Run a linear regression of sales on price_int, price_decimal1, price_decimal2, and feat. Report the coefficients of three price variables and explain how to interpret the coefficients.

In [ ]:
reg = smf.ols("sales ~ price_int + price_decimal1 + price_decimal2 + feat", data=df).fit()
print(reg.summary())

OLS Regression Results                            
Dep. Variable:                  sales   R-squared:                       0.407
Model:                            OLS   Adj. R-squared:                  0.406
Method:                 Least Squares   F-statistic:                     1652.
Date:                Tue, 18 Nov 2025   Prob (F-statistic):               0.00
Time:                        10:29:43   Log-Likelihood:            -1.0545e+05
No. Observations:                9649   AIC:                         2.109e+05
Df Residuals:                    9644   BIC:                         2.110e+05
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
Intercept       4.522e+04   1061.351     42.603      0.000    4.31e+04  

Interpretation of the coefficients:price_int: −13,620

A one-dollar increase in the whole-dollar part of the price is associated with a large drop in sales, averaging around 13,620 units.

price_decimal1: −11,780

Increasing the first decimal digit by 0.1 (for example from .8 to .9) also reduces sales, although by a smaller amount than changing the whole dollar.

price_decimal2: +41,490

The coefficient +41,490 is positive, which is somewhat counterintuitive since raising the price by a cent should normally reduce sales. An explintion for this could be that the model is picking up retail pricing patterns rather than a true causal effect of adding one cent.

Overall, the results show that higher prices reduce sales when looking at the whole-dollar and first-decimal parts of the price, which is consistent with normal demand behavior. The unusual positive effect on the second decimal digit likely reflects pricing conventions in the dataset rather than a real response to very small price changes.
(All statistically significant with p < 0.001.)

8. Combine the results from the tabulation and regression, discuss the nuanced patterns in consumer price sensitivity and the pricing strategy of Tropicana.

The tabulation shows that Tropicana prices rarely use the full range of decimal values. Instead, they cluster heavily at endings like .89 and .99, which explains why the first decimal digit concentrates around 0.8–0.9 and why the second decimal digit is almost always 0.09. These patterns indicate that Tropicana relies on narrow sets of price endings rather than uniform decimal pricing.
The regression results reinforce this. Consumers are strongly sensitive to changes in the whole-dollar and first-decimal parts of the price, with both coefficients large and negative. In contrast, the second decimal digit has a positive coefficient, which likely reflects the concentrated use of specific price endings rather than any real willingness by consumers to pay an extra cent. When certain decimal endings appear mostly in weeks with higher sales, the model attributes that pattern to the decimal digit itself.
Taken together, the price-ending distributions and the regression suggest that Tropicana uses highly targeted price endings and that consumers respond most to larger, more noticeable parts of the price. The small decimal digits mainly reflect the retailer’s pricing conventions rather than meaningful differences in consumer behavior.

Part III

9. Split the data into two parts. Run a linear regression of log(sales) on log(price) using part 1. Report the in-sample R2 for part 1 data and calculate the regression model’s out-of-sample R2 for part 2 data.

In [ ]:
from sklearn.model_selection import train_test_split
import numpy as np
import statsmodels.formula.api as smf

# split the data
df1, df2 = train_test_split(df, test_size=0.5, random_state=52610)
reg = smf.ols("np.log(sales) ~ np.log(price)", data=df1).fit()
in_samle_r2 = reg.rsquared
in_samle_r2

np.float64(0.4150809403518383)


In [ ]:
# Predict log(sales) for part 2
y_pred = reg.predict(df2)

# Compute true log(sales) for part 2
y_true = np.log(df2["sales"])

# Compute out-of-sample R2
sse = ((y_true - y_pred)**2).sum()
sst = ((y_true - y_true.mean())**2).sum()
out_of_sample_r2 = 1 - sse/sst
out_of_sample_r2

np.float64(0.43822604660735254)


In-sample R2 for part 1: 0.415080940351838
Out-of-sample R2 for part 2: 0.43822604660735254

10. Compare the in-sample and out-of-sample R2, and discuss the result.

The in-sample R² is 0.415080940351838 and the out-of-sample R² is 0.43822604660735254. The two values are very close, which indicates that the simple log–log model fits both halves of the data in a similar way. The slightly higher out-of-sample R² is reasonable in a large dataset where the training and testing subsets come from the same underlying distribution. Overall, the model shows stable performance and does not appear to overfit.